libcxx/src/new.cpp

```diff
@@ -54,7 +54,7 @@ _LIBCPP_WEAK void* operator new(size_t size, const std::nothrow_t&) noexcept {
 #  ifdef _LIBCPP_HAS_NO_EXCEPTIONS
 #    if _LIBCPP_CAN_DETECT_OVERRIDDEN_FUNCTION
   _LIBCPP_ASSERT_SHIM(
-      !std::__is_function_overridden(static_cast<void* (*)(std::size_t)>(&operator new)),
+      !std::__is_function_overridden<std::__overload_of<void* (std::size_t)>(&operator new)>(),
```

Comment:
What's the benefit of using `__overload_of` here instead of just `static_cast`?

Reply:
Clang doesn't like `static_cast`:
```
error: address of overloaded function 'operator new' cannot be static_cast to type 'void *(size_t)' (aka 'void *(unsigned long)')
...
error: address of overloaded function 'operator new[]' cannot be static_cast to type 'void *(size_t)' (aka 'void *(unsigned long)')
```

Reply:
Thanks, I made this change and Clang seems to be happy with it.